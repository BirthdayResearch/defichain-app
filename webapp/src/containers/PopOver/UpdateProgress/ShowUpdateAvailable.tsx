import React from 'react';
import { Button, ModalBody, ModalFooter } from 'reactstrap';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { showAvailableUpdateResponse } from '../../../app/update.ipcRenderer';
import { closeUpdateAvailable } from '../reducer';

interface ShowUpdateAvailableComponentProps {
  closeUpdateAvailable: () => void;
  closeModal: (fn: any) => void;
}

const ShowUpdateAvailableComponent = (
  props: ShowUpdateAvailableComponentProps
) => {
  const { closeUpdateAvailable, closeModal } = props;
  const closing = () => closeModal(closeUpdateAvailable);
  return (
    <>
      <ModalBody>
        <h1 className='h4'>{I18n.t('alerts.showUpdateAvailableHeader')}</h1>
        {I18n.t('alerts.showUpdateAvailableNotice')}
      </ModalBody>
      <ModalFooter>
        <Button size='sm' color='link' onClick={closing}>
          {I18n.t('alerts.noShowUpdateAvailableButton')}
        </Button>
        <Button size='sm' color='primary' onClick={showAvailableUpdateResponse}>
          {I18n.t('alerts.yesShowUpdateAvailableButton')}
        </Button>
      </ModalFooter>
    </>
  );
};

const mapDispatchToProps = {
  closeUpdateAvailable,
};

export default connect(null, mapDispatchToProps)(ShowUpdateAvailableComponent);
