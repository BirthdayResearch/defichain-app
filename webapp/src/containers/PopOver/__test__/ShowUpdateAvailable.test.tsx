import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import ShowUpdateAvailable from '../UpdateProgress/ShowUpdateAvailable';
import Enzyme from '../../../utils/testUtils/enzyme';

describe('ShowUpdateAvailable', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <ShowUpdateAvailable closeModal={() => {}} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
