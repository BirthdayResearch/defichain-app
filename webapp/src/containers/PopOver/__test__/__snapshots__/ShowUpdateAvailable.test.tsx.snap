// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowUpdateAvailable should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ShowUpdateAvailableComponent)
    closeModal={[Function]}
  >
    <ShowUpdateAvailableComponent
      closeModal={[Function]}
      closeUpdateAvailable={[Function]}
    >
      <ModalBody
        tag="div"
      >
        <div
          className="modal-body"
        >
          <h1
            className="h4"
          >
            An update is available for DeFi Wallet
          </h1>
          Would you like to update?
        </div>
      </ModalBody>
      <ModalFooter
        tag="div"
      >
        <div
          className="modal-footer"
        >
          <Button
            color="link"
            onClick={[Function]}
            size="sm"
            tag="button"
          >
            <button
              aria-label={null}
              className="btn btn-link btn-sm"
              onClick={[Function]}
              type="button"
            >
              Not now
            </button>
          </Button>
          <Button
            color="primary"
            onClick={[Function]}
            size="sm"
            tag="button"
          >
            <button
              aria-label={null}
              className="btn btn-primary btn-sm"
              onClick={[Function]}
              type="button"
            >
              Update
            </button>
          </Button>
        </div>
      </ModalFooter>
    </ShowUpdateAvailableComponent>
  </Connect(ShowUpdateAvailableComponent)>
</Provider>
`;
